src/qamqpclient.cpp

```diff
@@ -660,14 +660,20 @@ QAmqpExchange *QAmqpClient::createExchange(int channelNumber)
 QAmqpExchange *QAmqpClient::createExchange(const QString &name, int channelNumber)
 {
     Q_D(QAmqpClient);
+    if (!name.isEmpty() && d->exchanges.contains(name)
+                    && (!d->exchanges[name].isNull()))
```

Comment:
I'm just looking at `QObjectCleanupHandler`, is that something we're supposed to subclass?  Seems you can put object pointers into it, take them out, see if there are any left, but never actually see them.

If it could manage the contents of a `QHash` or `QMap`, then it could be quite useful.  Google doesn't seem to know of too many people using it.

In the meantime I've just put an update out that does a garbage collection run on receipt of a `destroyed` signal.  So that in theory takes care of the memory leak.  I was thinking about maybe having a delay there so that if a few objects get deleted, it just runs once and collects the lot of them.

Then again, how often are exchanges and queues removed?  I'd think this is going to be a less likely scenario.  One that needs handling for sure, but not one that needs a highly optimised solution.